core/pkg/server/sender.go

```diff
@@ -1060,7 +1060,7 @@ func (s *Sender) upsertRun(record *spb.Record, run *spb.RunRecord) {
 	data, err := gql.UpsertBucket(
 		ctx,                                // ctx
 		s.graphqlClient,                    // client
-		nil,                                // id
+		nullify.NilIfZero(run.StorageId),   // id, required to avoid writes by users without perms
```

Comment:
the explanation doesn't make sense, as this value could be nil, please add proper explanation, is it only relevant when resuming a run?